From 241d2e765dc7401e642812e43b75dbc3950f2c98 Mon Sep 17 00:00:00 2001
From: Mark Rousskov <mark.simulacrum@gmail.com>
Date: Wed, 18 Dec 2019 13:28:14 -0500
Subject: [PATCH] Fix compiletest fallout from stage0 bump

---
 src/tools/compiletest/src/main.rs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/tools/compiletest/src/main.rs b/src/tools/compiletest/src/main.rs
index 15f8abd75d59..32965bbb292d 100644
--- a/src/tools/compiletest/src/main.rs
+++ b/src/tools/compiletest/src/main.rs
@@ -569,6 +569,7 @@ pub fn test_opts(config: &Config) -> test::TestOpts {
         list: false,
         options: test::Options::new(),
         time_options: None,
+        force_run_in_process: false,
     }
 }
 
-- 
2.24.1

